spec/controllers/dashboard_controller_spec.rb

```diff
@@ -208,27 +208,33 @@
   end
 
   context "Create Dashboard" do
+    let(:widget) { FactoryBot.create(:miq_widget) }
+
     before do
+      EvmSpecHelper.local_miq_server
       @group = FactoryBot.create(:miq_group, :miq_user_role => FactoryBot.create(:miq_user_role, :features => %w[everything]))
       @user = FactoryBot.create(:user, :miq_groups => [@group])
       # create dashboard for a group
       @ws = FactoryBot.create(:miq_widget_set,
                               :name     => "group_default",
-                              :set_data => {:last_group_db_updated => Time.now.utc,
-                                            :col1 => [], :col2 => [], :col3 => []},
+                              :set_data => {:col1 => [widget.id], :col2 => [], :col3 => [], :last_group_db_updated => Time.now.utc},
                               # :userid   => @user.userid,
-                              :group_id => @group.id)
+                              :owner    => @group)
       @group.update(:settings => {:dashboard_order => [@ws.id]})
     end
 
     it "dashboard show" do
+      EvmSpecHelper.local_miq_server
       controller.instance_variable_set(:@sb, :active_db => @ws.name)
       controller.instance_variable_set(:@tabs, [])
       login_as @user
       # create a user's dashboard using group dashboard name.
       FactoryBot.create(:miq_widget_set,
-                        :name     => "#{@user.userid}|#{@group.id}|#{@ws.name}",
-                        :set_data => {:last_group_db_updated => Time.now.utc, :col1 => [1], :col2 => [], :col3 => []})
+                        :name     => @user.userid.to_s,
+                        :owner    => @user,
+                        :group_id => @user.current_group_id,
```

Comment:
Does this mean we should fix the before validation in the model to work with users?
Thinking having a `user_with_group` may be what we want